Add clear button to action menus

Maxador opened the pull request:
This add a clear button as the left nav bar item for the `PeopleViewController`, `LabelsViewController` and the `MilestonesViewController`.
To keep the selected count in the `PeopleViewController`, I moved the counter in the title next to the "Assignees" or "Reviewers" title text.

#2288

Huddie commented:
@Maxador a screenshot of what it looks like would be helpful 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

rnystrom commented on Classes/Labels/LabelsViewController.swift:
```diff
@@ -10,7 +10,13 @@ import UIKit
-final class LabelsViewController: BaseListViewController2<String>, BaseListViewController2DataSource {
+protocol LabelsViewControllerDelegate {
+    func didSelect()
+}
```
Can this protocol be removed? Looks unused.

Maxador replied:
Yes it is an artifact of my development, I will remove it.

rnystrom commented on Classes/People/PeopleViewController.swift:
```diff
@@ -90,17 +109,16 @@ PeopleSectionControllerDelegate {
+        let selectedCount = "\(selected.count)/\(selectionLimit)"
+        switch type {
+        case .assignee: title = "\(NSLocalizedString("Assignees", comment: "")) \(selectedCount)"
+        case .reviewer: title = "\(NSLocalizedString("Reviewers", comment: "")) \(selectedCount)"
```
I think we should keep the selection count formatting b/c there is a maximum for selected people (10)

rnystrom commented on Classes/Milestones/MilestonesViewController.swift:
```diff
@@ -55,6 +59,27 @@ MilestoneSectionControllerDelegate {
+
+    func addMenuClearButton() {
+        navigationItem.leftBarButtonItem = UIBarButtonItem(
+            title: NSLocalizedString("Clear", comment: ""),
```
Can we add this string to the `Constants.Strings` list since its used multiple times here?

Maxador commented:
Here are the screenshots:
![img_1825](https://user-images.githubusercontent.com/1437036/46918473-ada83880-cfa0-11e8-9440-6f595e9e699b.png)
![img_1824](https://user-images.githubusercontent.com/1437036/46918474-aed96580-cfa0-11e8-8bde-2e273a9d7ed9.png)

The counter fits in the title for the `PeopleViewController` but it is a little tight. Since it is a really small view, I don't know if adding some kind of header or footer (or even a toolbar) to always show a clear button or the counter is a good idea.

One idea I had is to show a `Clear?` label when we tap on the counter (on top left like before). And if we tap again, the clear happen. Otherwise, after a short delay, the counter show up again. While this clear some space in the view, it makes the feature less discoverable.

rnystrom commented:
Ohh I see, duh, the left bar item is used for the clear button. Hmmm. Ok maybe we just go with putting it in the title for now.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

Maxador commented:
I added the constants and fixed a bug I noticed. The Clear button state (enable or not) and the `PeopleViewController` title wasn't set to the correct initial selection value.

Maxador commented:
I think we can close #1942 now.
